sdks/python/apache_beam/runners/direct/executor.py

```diff
@@ -602,7 +599,17 @@ def _fire_timers(self):
       return bool(transform_fired_timers)
 
     def _is_executing(self):
-      """Returns True if there is at least one non-blocked TransformExecutor."""
+      """Checks whether the job is still executing.
+
+      Returns:
+        True if there are any timers set or if there is at least
+        one non-blocked TransformExecutor active."""
+
+      watermark_manager = self._executor.evaluation_context._watermark_manager
+      _, any_unfired_timers = watermark_manager.extract_all_timers()
+      if any_unfired_timers:
```

Comment:
<!--new_thread; commit:c6bc606e32be4f68c073c7916767666d2e91a977; resolved:0-->
Can you more clearly indicate (by naming or comment) that this is for unfired real time timers?